testutil/proof/fixture_generators.go

```diff
@@ -30,11 +31,24 @@ func BaseClaim(appAddr, supplierAddr string, sum uint64) prooftypes.Claim {
 	}
 }
 
-// SmstRootWithSum returns a SMST root with the given sum that can be used for
-// testing.
+// SmstRootWithSum returns a SMST root with the given sum and a default
+// hard-coded count of 1.
+// TODO_POTENTIAL_TECHDEBT: Note that the count is meant to represent the number
+// of non-empty leaves in the tree, and may need become a parameter depending on
+// how the tests evolve.
 func SmstRootWithSum(sum uint64) smt.MerkleRoot {
-	root := make([]byte, 40)
-	copy(root[:32], []byte("This is exactly 32 characters!!!"))
-	binary.BigEndian.PutUint64(root[32:], sum)
+	root := make([]byte, smt.SmstRootSizeBytes)
+
+	// Insert a string into the root hash
+	copy(root[:smt.SmtRootSizeBytes], []byte(fmt.Sprintf("This is exactly %d characters!!!", smt.SmtRootSizeBytes)))
+
+	// Insert the sum into the root hash
+	binary.BigEndian.PutUint64(root[smt.SmtRootSizeBytes:], sum)
+
+	// Insert the count into the root hash
+	// TODO_TECHDEBT: This is a hard-coded count of 1, but could be a parameter.
+	// TODO_TECHDEBT: We are assuming the sum takes up 8 bytes.
+	binary.BigEndian.PutUint64(root[smt.SmtRootSizeBytes+8:], 1)
+
 	return smt.MerkleRoot(root)
```

Comment:
1. ~~Why the `+8` while slicing `root` in the call to `binary.BigEndian.PutUint64`? It seems to me that this should cause an index out of range error. See: [this go playground example](https://go.dev/play/p/VDtFoFpfo09).~~

    I'm noticing that the distinction between `smt.SmtRootSizeBytes` and `smt.SmstRootSizeBytes` is *very* subtle and easy to miss when reading this.

1. Is there a reason that we're using big endian? My understanding of endianness is that little endian is conventional in most contexts, networking being the common exception.

1. Do we actually need to include any data in the root? If not, why not do something like this:

```suggestion
	root := [smt.SmstRootSizeBytes]byte{}

	// Insert the sum into the root hash
	binary.BigEndian.PutUint64(root[smt.SmtRootSizeBytes:], sum)

	// Insert the count into the root hash
	// TODO_TECHDEBT: This is a hard-coded count of 1, but could be a parameter.
	// TODO_TECHDEBT: We are assuming the sum takes up 8 bytes.
	binary.BigEndian.PutUint64(root[smt.SmtRootSizeBytes+8:], 1)

	return smt.MerkleRoot(root[:])
```